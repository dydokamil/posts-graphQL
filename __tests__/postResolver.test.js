const { Query, Mutation } = require('../resolvers')
const { graphql } = require('graphql')
const mongoose = require('mongoose')

const schema = require('../schema')
const Post = require('../models/post')
const User = require('../models/user')
const { MONGO_URL_DEV } = require('../consts')

describe('post resolver', () => {
  const userData = {
    _id: '5abba8e47af4d91c259e12ef',
    username: 'User1',
    email: 'User1@gql.com',
    createdAt: '2018-12-12T13:00:00',
    lastLogin: '2018-12-12T03:14:07',
    posts: ['41224d776a326fb40f000002']
  }

  const postData = {
    _id: '5abba8e47af4d91c259e12ee',
    author: userData._id,
    createdAt: '2018-10-10T13:00:00',
    editedAt: '2018-10-10T13:00:00',
    message: 'Some message'
  }

  beforeAll(async () => {
    await mongoose.connect(MONGO_URL_DEV)
    // clear the database
    await Post.removePosts()
    await User.removeUsers()
  })

  beforeEach(async () => {
    const userInstance = new User(userData)
    await userInstance.save()

    const postInstance = new Post(postData)
    await postInstance.save()
  })

  afterEach(async () => {
    await Post.removePosts()
    await User.removeUsers()
  })

  afterAll(done => {
    mongoose.disconnect(done)
  })
  it('should get a list of posts', async () => {
    const posts = await Query.posts()
    expect(posts.length).toBe(1)
    expect(posts).toMatchSnapshot()
  })

  it('should get a specific post', async () => {
    const { _id } = postData
    const user = await Query.post({}, { _id })
    expect(user._id.equals(_id)).toBeTruthy()
    expect(user).toMatchSnapshot()
  })

  it('should create a post', async () => {
    const message = 'Hello world!'
    const author = userData._id
    const createPost = await Mutation.createPost({}, { message, author })
    expect(createPost.message).toBe(message)
  })
})
