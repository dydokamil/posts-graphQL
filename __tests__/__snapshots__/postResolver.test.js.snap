// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post resolver should get a list of posts 1`] = `
Array [
  Object {
    "__v": 0,
    "_id": "5abba8e47af4d91c259e12ee",
    "author": null,
    "createdAt": "2018-10-10T13:00:00",
    "editedAt": "2018-10-10T13:00:00",
    "message": "Some message",
  },
]
`;

exports[`post resolver should get a specific post 1`] = `
Object {
  "__v": 0,
  "_id": "5abba8e47af4d91c259e12ee",
  "author": null,
  "createdAt": "2018-10-10T13:00:00",
  "editedAt": "2018-10-10T13:00:00",
  "message": "Some message",
}
`;
